ibind/base/rest_client.py

```diff
@@ -147,8 +187,21 @@ def request(self, method: str, endpoint: str, attempt: int = 0, log: bool = True
             Exception: For any other errors that occur during the request.
 
         """
+
+        # TODO: raise error if use_oauth=False, but live_session_token=None
+
+        if self._use_oauth:
+            base_url="https://api.ibkr.com/v1/api/"
+        else:
+            base_url = base_url if base_url is not None else self.base_url
```

Comment:
Do I understand correctly that when we use OAuth all requests have to be directed at this base url?

Reply:
Yes, the base url for oauth is different. There is probably a better method, but this got the code working.

Reply:
Riiiiiight! I never realised this! I think I made a few changes and comments based on an incorrect assumption that only getting live tokens requires this base URL, while all the usual endpoints use the original one. I'm so sorry, that must probably have made your life harder.

Sorry, then in this case we should tackle this differently. We'll read a different base url from env vars when IbkrClient gets constructed.

----

First, let's duplicate this variable and change it to IBIND_OAUTH_REST_URL:

https://github.com/Voyz/ibind/blob/ce808fb3463b5e6306474ade47088dae0147702b/ibind/var.py#L50-L51

```python
IBIND_OAUTH_REST_URL= os.getenv('IBIND_OAUTH_REST_URL', 'https://api.ibkr.com/v1/api/')
""" IBKR Client Portal Gateway's URL for OAuth REST API."""
```

----

Here:
https://github.com/Voyz/ibind/blob/ce808fb3463b5e6306474ade47088dae0147702b/ibind/client/ibkr_client.py#L41

Let's change it to default to None.

----

Finally:

https://github.com/Voyz/ibind/blob/ce808fb3463b5e6306474ade47088dae0147702b/ibind/client/ibkr_client.py#L62-L68

Let's define the url differently depending on the `use_oauth` parameter:
```python
self._use_oauth = use_oauth

url = var.IBIND_OAUTH_REST_URL if self._use_oauth else var.IBIND_REST_URL
    
if url is None:
    url = f'https://{host}:{port}{base_route}'

self.account_id = account_id
super().__init__(url=url, cacert=cacert, timeout=timeout, max_retries=max_retries)
```

----

Then as a result, this:

https://github.com/Voyz/ibind/blob/ce808fb3463b5e6306474ade47088dae0147702b/ibind/base/rest_client.py#L192-L195

Can be restored back to:

```python
base_url = base_url if base_url is not None else self.base_url
```

And RestClient once again no longer will need `use_oauth` member field - it will only live in IbkrClient.

Once again, sorry for the confusion and thanks for your patience 🙏

Reply:
Yes! That caused a few problems when I merged your changes without checking them properly, and I had to find the bugs. No problem though.

I'll work through the changes and commit when done and working.